remotes/docker/pusher.go

```diff
@@ -132,6 +145,48 @@ func (p dockerPusher) Push(ctx context.Context, desc ocispec.Descriptor) (conten
 		}
 		req.Header.Add("Content-Type", desc.MediaType)
 	} else {
+		mountCandidates := findMatchingOrigins(p.refspec, p.originProvider(desc))
+		for _, mountCandidate := range mountCandidates {
+			ctx, err := contextWithRepositoryScope(ctx, mountCandidate, false, true)
+			if err != nil {
+				// try next candidate
+				continue
+			}
+			i := strings.Index(mountCandidate.Locator, "/")
+			originRepoPath := mountCandidate.Locator[i+1:]
+			req, err = http.NewRequest(http.MethodPost, fmt.Sprintf("%s/?mount=%s&from=%s", p.url("blobs", "uploads"), desc.Digest, originRepoPath), nil)
+			if err != nil {
+				// try next candidate
+				continue
+			}
+
+			resp, err = p.doRequestWithRetries(ctx, req, nil)
+			if err != nil {
+				// try next candidate
+				continue
+			}
+
+			switch resp.StatusCode {
+			case http.StatusCreated:
+			default:
+				// try next candidate
+				continue
+			}
+
+			p.tracker.SetStatus(ref, Status{
+				Status: content.Status{
+					Ref:       ref,
+					Total:     desc.Size,
+					Expected:  desc.Digest,
+					StartedAt: time.Now(),
+					Offset:    desc.Size,
+				},
+			})
+			// mount succeeded, returning a nil writer, without error
+			return nil, errors.Wrapf(errdefs.ErrAlreadyExists, "content %v mounted from %s", desc.Digest, mountCandidate)
```

Comment:
I know what you mean, but technically this is returning with an error. Maybe say "with canary error value"?

I also think this sort of subtlety is worthy of a godoc comment on the function itself.